lib/train/transports/ssh_connection.rb

```diff
@@ -273,18 +273,20 @@ def to_s
     #         not received.
     #
     # @api private
-    def execute_on_channel(cmd, &data_handler)
+    def execute_on_channel(cmd, **opts, &data_handler)
       stdout = stderr = ""
       exit_status = nil
       session.open_channel do |channel|
         # wrap commands if that is configured
         cmd = @cmd_wrapper.run(cmd) unless @cmd_wrapper.nil?
 
-        if @transport_options[:pty]
+        # In case of Windows SSH we are checking opts[:pty] flag
+        if opts[:pty].nil? ? opts[:pty] : @transport_options[:pty]
```

Comment:
This logic looks backwards to me. If `opts[:pty]` is nil, use it, otherwise use `transpart_options`... I think it should be:

```ruby
if opts[:pty] || @transport_options[:pty]
```

Unless we're trying to use trinary logic (which I'll want to shoot down as well, but separately)? Am I misunderstanding something?

Reply:
Thank you @zenspider. You are right in a hurry I have accidentally removed !. 
Second we have to using ternary logic as we want to overwrite default @transport_options[:pty] flag and use that in the if condition ahead. Hope this answers your queries.